### ISSUE TITLE
[opencv4:x64-windows] build failure

### ISSUE THREAD

**Issue by diegohdezr** [microsoft/vcpkg#11452]
**Host Environment**
- OS: windows 10 Version 10.0.18363.836
- Compiler: revision: 19.25.28614

**To Reproduce**
.\vcpkg.exe install opencv[core,vtk,contrib,nonfree]:x64-windows

**Failure logs** 
```
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: ninja -v
    Working Directory: C:/dev/vcpkg/buildtrees/opencv4/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\dev\vcpkg\buildtrees\opencv4\config-x64-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:297 (vcpkg_execute_required_process)
  ports/opencv4/portfile.cmake:262 (vcpkg_configure_cmake)
  scripts/ports.cmake:90 (include)


Error: Building package opencv4:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: opencv4:x64-windows
  Vcpkg version: 2020.02.04-nohash
```
**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.
I already updatet to the latest windows SDK available in visual studio installer
formated the PC

output from CMAKE error output log:
```
Determining if the include file C:/dev/vcpkg/installed/x64-windows/include/libpng/png.h exists failed with the following output:
Change Dir: C:/dev/vcpkg/buildtrees/opencv4/x64-windows-dbg/CMakeFiles/CMakeTmp

Run Build Command(s):C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe cmTC_9e023 
[1/2] Building C object CMakeFiles\cmTC_9e023.dir\CheckIncludeFile.c.obj

FAILED: CMakeFiles/cmTC_9e023.dir/CheckIncludeFile.c.obj 

C:\PROGRA~2\MICROS~2\2019\COMMUN~1\VC\Tools\MSVC\1425~1.286\bin\Hostx64\x64\cl.exe  /nologo   /DWIN32 /D_WINDOWS /W3 /utf-8 /MP   /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:precise /FS       /MP8   /MDd /Zi /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\cmTC_9e023.dir\CheckIncludeFile.c.obj /FdCMakeFiles\cmTC_9e023.dir\ /FS -c CheckIncludeFile.c

CheckIncludeFile.c(1): fatal error C1083: Cannot open include file: 'C:/dev/vcpkg/installed/x64-windows/include/libpng/png.h': No such file or directory
ninja: build stopped: subcommand failed.
```

Determining if the include file pthread.h exists failed with the following output:
```
Change Dir: C:/dev/vcpkg/buildtrees/opencv4/x64-windows-dbg/CMakeFiles/CMakeTmp

Run Build Command(s):C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe cmTC_ab813 
[1/2] Building C object CMakeFiles\cmTC_ab813.dir\CheckIncludeFile.c.obj

FAILED: CMakeFiles/cmTC_ab813.dir/CheckIncludeFile.c.obj 

C:\PROGRA~2\MICROS~2\2019\COMMUN~1\VC\Tools\MSVC\1425~1.286\bin\Hostx64\x64\cl.exe  /nologo   /DWIN32 /D_WINDOWS /W3 /utf-8 /MP   /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:precise /FS       /MP8   /MDd /Zi /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\cmTC_ab813.dir\CheckIncludeFile.c.obj /FdCMakeFiles\cmTC_ab813.dir\ /FS -c CheckIncludeFile.c

CheckIncludeFile.c(1): fatal error C1083: Cannot open include file: 'pthread.h': No such file or directory
ninja: build stopped: subcommand failed.
```

**Comment by JackBoosY** [comment]
Hope # 11130 will solve this issue.

**Comment by cenit** [comment]
Please post full logs. Extracts like fatal error C1083: Cannot open include file: 'pthread.h': No such file or directory unfortunately are not very helpful and usually are just the last step of a cascade of failures

**Comment by diegohdezr** [comment]
hello, here are the logs. Should I upload the CMake logs too?

[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/4658984/config-x64-windows-out.log)

**Comment by diegohdezr** [comment]
one thing to add that may be of interest, the same command works fine in a desktop pc without issue, but fails in a laptop, I tried with two desktops and two laptops and it only failed on the laptops

**Comment by JackBoosY** [comment]
Hi @diegohdezr, since #11130 merged, does this issue still exist?

**Comment by guybashan** [comment]
I am having the same issue.. Running on a Desktop..

**Comment by JackBoosY** [comment]
@guybashan can you provide me the full logs?

Thanks.

**Comment by guybashan** [comment]
Attached.
Please let me know if there are any other logs of interest.

[CMakeOutput.log](https://github.com/microsoft/vcpkg/files/5453591/CMakeOutput.log)
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/5453592/config-x64-windows-out.log)
[CMakeError.log](https://github.com/microsoft/vcpkg/files/5453593/CMakeError.log)

**Comment by JackBoosY** [comment]
@guybashan Oh, this issue should be fixed in #14177, please track issue #14171.
